lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	ManifestRepo string
```

Comment:
Probably `GitRepoURL` is better. Even better if it's not a `string` and actualy a URL type (see https://golang.org/pkg/net/url/#URL).

Reply:
The function I use GitRepoURL for (gogit.PlainOpen) actually expects a string directory path

Reply:
Even so, it's better to not use strings for any internal data representation. You can easily convert a URL to a string if a library function expects a string, but it does not mean that we should also store strings for ourselves.